lms/djangoapps/courseware/access.py

```diff
@@ -370,11 +400,12 @@ def _can_load_course_overview(user, course_overview):
         The user doesn't have to be enrolled in the course in order to have load
         load access.
     """
-    return (
-        not course_overview.visible_to_staff_only
+    response = (
+        _visible_to_nonstaff_users(course_overview)
         and _can_access_descriptor_with_start_date(user, course_overview, course_overview.id)
-    ) or _has_staff_access_to_descriptor(user, course_overview, course_overview.id)
+    )
 
+    return response if response else _has_staff_access_to_descriptor(user, course_overview, course_overview.id)
```

Comment:
Same as the other one.  You're probably getting the incorrect error object, as you had explained to me earlier.